packages/block-library/src/navigation-link/fallback-variations.js

```diff
@@ -58,8 +58,14 @@ const fallbackVariations = [
  */
 fallbackVariations.forEach( ( variation ) => {
 	if ( variation.isActive ) return;
-	variation.isActive = ( blockAttributes, variationAttributes ) =>
-		blockAttributes.type === variationAttributes.type;
+
+	variation.isActive = ( blockAttributes, variationAttributes ) => {
+		const normalizedVariationKind = variationAttributes.kind.replace(
+			'-',
+			'_'
+		);
+		return blockAttributes.type === normalizedVariationKind;
+	};
```

Comment:
I'm not sure comparing the kind to the type like this will work (e.g. for a post the kind is `postType` and the type `post`).

What was the problem that required this change?

Reply:
Thanks for calling this out. Turns out I needed to be storing/retrieveing data from the `object` field instead. With that change we can now revert both the variations `isActive` comparisons to how they were previously.